packages/design-system/src/components/web-components/ds-date-field/ds-date-field.stories.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from 'react';
+import { action } from '@storybook/addon-actions';
+import type { Meta } from '@storybook/react';
+import WebComponentDocTemplate from '../../../../../../.storybook/docs/WebComponentDocTemplate.mdx';
+import { webComponentDecorator } from '../storybook';
+import './ds-date-field';
+
+const meta: Meta = {
+  title: 'Web Components/ds-date-field',
+  decorators: [webComponentDecorator],
+  parameters: {
+    docs: {
+      page: WebComponentDocTemplate,
+    },
+  },
+  args: {
+    hint: 'If you were born on a leap day, entering the date will either crash our servers or open a portal to an alternate dimension.',
+    label: 'Enter your date of birth.',
+    name: 'ds-date-field',
+  },
+  argTypes: {
+    errorMessage: { control: 'text' },
+    hint: { control: 'text' },
+    requirementLabel: { control: 'text' },
+  },
+};
+export default meta;
+
+const Template = (args) => {
+  useEffect(() => {
+    const element = document.querySelector('ds-date-field');
+    if (element) {
+      const handleStoryBookChange = (event: CustomEvent) => {
```

Comment:
We know what our custom event's `detail` type will be, so we can pass that into the `CustomEvent` interface, like so:

```suggestion
      const handleStoryBookChange = (event: CustomEvent<{ updatedValue: string; formattedValue: string }> ) => {
```

Two additional resources:
- an [example](https://stackoverflow.com/a/62568282)
- link to the [typescript definition](https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/29de1d981d83c2e99f6995c4e5fae934cd7d709a/baselines/audioworklet.generated.d.ts#L306).